storage/s3_sdk/list_gcs_buckets_test.go

```diff
@@ -17,29 +17,78 @@ package s3sdk
 
 import (
 	"bytes"
+	"context"
+	"fmt"
 	"io/ioutil"
 	"log"
 	"os"
 	"strings"
 	"testing"
+
+	"cloud.google.com/go/storage"
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
 )
 
+var serviceAccountEmail = os.Getenv("GOLANG_SAMPLES_SERVICE_ACCOUNT_EMAIL")
+var storageClient *storage.Client
+
 func TestMain(m *testing.M) {
 	log.SetOutput(ioutil.Discard)
 	s := m.Run()
 	log.SetOutput(os.Stderr)
+
+	ctx := context.Background()
+	storageClient, _ = storage.NewClient(ctx)
+	defer storageClient.Close()
+
+	tc, _ := testutil.ContextMain(m)
+	key, err := createTestKey(tc.ProjectID)
+	if err != nil {
+		log.Fatalf("error setting up HMAC key: %v", err)
+	}
+	defer deleteTestKey(key)
+
 	os.Exit(s)
 }
 
+// Create a key for testing purposes and set environment variables
+func createTestKey(projectID string) (*storage.HMACKey, error) {
```

Comment:
Is there any sort of limit on the number of keys? If we delete, does it delete forever or go into a "disabled" state that still counts against a limit?

We've had issues with other key type resources where they don't/can't get completely cleaned up and we run into a limit.